### PR TITLE
Start up server HTTP interface earlier

### PR DESCRIPTION
**Pull request by jsteemann**
Upstream PRs:
- https://github.com/arangodb/arangodb/pull/16181
- https://github.com/arangodb/arangodb/pull/16381

TODO:
- [x] Shorten release notes
- [x] Update [DocuBlock](https://github.com/arangodb/arangodb/blob/devel/Documentation/DocuBlocks/Rest/Administration/JSF_get_admin_status.md) (needs review from Kaveh)
- [x] Merge the open upstream PR and update this PR's feature branch before re-generating the examples

### CODE REVIEW

File: 3.10/release-notes-api-changes310.md [hunk 1]
```diff
@@ -46,6 +104,31 @@ The following HTTP APIs are affected:
...
+   stopped or unprepared.
+- `recoveryTick`: current recovery sequence number value if the instance is currently in
+  recovery. If the instance is already past the recovery, this attribute will contain 
+  the last handled recovery sequence number.
+
+The exact values of these attributes should not be relied on, i.e. client applications
+should not check for any exact values in them. Feature and phase names are subject to
+change between different versions of ArangoDB. 
+The progress attributes can still be used to determine if the instance has made progress
+between two calls: if `phase`, feature` and `recoveryTick` don't change, then there hasn't
```
Comment (neunhoef) [inline 3.10/release-notes-api-changes310.md:124]:
```suggestion
between two calls: if `phase`, `feature` and `recoveryTick` don't change, then there hasn't
```

**Review by neunhoef — APPROVED** [review]
LGTM. One typo suggested to correct.

File: 3.10/http/general.md [hunk 2]
```diff
@@ -611,3 +611,41 @@ receiving Coordinator will also check the header on its own.
 Apart from that, the header will not be included in cluster-internal requests
 executed by the Coordinator, e.g. when the Coordinator issues sub-requests
 to DB-Servers or Agency instances.
+
+Responding to liveliness probes
+-------------------------------
+
+You can now optionally start the HTTP REST interface of arangod instances early or late 
+during the startup process. ArangoDB versions before 3.10 only allowed starting the interface
+late, after the recovery was finished.
```
Comment (Simran-B) [inline 3.10/http/general.md:620]: It's not necessarily clear to users what the recovery is and that it's always run, not just after a crash.

File: 3.10/release-notes-api-changes310.md [hunk 3]
```diff
@@ -11,6 +11,64 @@ integrations for ArangoDB 3.10.
...
```
Comment (Simran-B) [inline 3.10/release-notes-api-changes310.md:64]:
```suggestion
between two calls: if `phase`, `feature` and `recoveryTick` don't change, then there hasn't
```

File: 3.10/release-notes-api-changes310.md [hunk 4]
```diff
@@ -46,6 +104,31 @@ The following HTTP APIs are affected:
 * `/_api/wal/tail`
 * `/_api/replication/logger-follow`
 
+
+The GET `/_admin/status` API now also returns startup and recovery information. This
+can be used to determine the instance's progress during startup. The new `progress`
+attribute will be returned inside the `serverInfo` object with the following subattributes:
+
+- `phase`: name of the lifecycle phase the instance is currently in. Normally one of
+  "in prepare", "in start", "in wait", "in shutdown", "in stop" or "in unprepare".
```
Comment (Simran-B) [inline 3.10/release-notes-api-changes310.md:113]:
```suggestion
  `"in prepare"`, `"in start"`, `"in wait"`, `"in shutdown"`, `"in stop"`, or `"in unprepare"`.
```

File: 3.10/release-notes-api-changes310.md [hunk 5]
```diff
@@ -11,6 +11,64 @@ integrations for ArangoDB 3.10.
...
+require access to the database data (e.g. HTTP basic authentication) will also be 
+responded to with HTTP 503 errors, even if correct credentials are used. This is
+because access to the database data is not possible early during the startup.
+
+The GET `/_admin/status` API now also returns startup and recovery information. This
+can be used to determine the instance's progress during startup. The new `progress`
+attribute will be returned inside the `serverInfo` object with the following subattributes:
+
+- `phase`: name of the lifecycle phase the instance is currently in. Normally one of
+  "in prepare", "in start", "in wait", "in shutdown", "in stop" or "in unprepare".
```
Comment (Simran-B) [inline 3.10/release-notes-api-changes310.md:53]:
```suggestion
  `"in prepare"`, `"in start"`, `"in wait"`, `"in shutdown"`, `"in stop"`, or `"in unprepare"`.
```

File: 3.10/release-notes-api-changes310.md (hunk 3 above)
Comment (Simran-B) [inline 3.10/release-notes-api-changes310.md:64]:
```suggestion
between two calls: if `phase`, `feature` and `recoveryTick` don't change, then there hasn't
```

**Review by Simran-B — CHANGES_REQUESTED** [review]
The last bullet point needs to be adjusted accordingly:
https://github.com/arangodb/docs/blob/feature/open-http-interface-earlier/3.10/http/general.md#error-handling

**Comment by Simran-B** [comment]
@joerg84 I'm not sure what you try to say. This PR is blocked by https://github.com/arangodb/arangodb/pull/16381 as indicated by the updated description.

**Comment by ansoboleva** [comment]
https://jenkins.arangodb.biz:3456/view/Documentation/job/arangodb-ANY-examples/2132/

**Pull request merged**